File: src/app/demo/pages/medicamento/models/medicamento.ts
export class Medicamento {
  id?: number;
  nombre!: string;
  descripcion!: string;
  presentacion!: string;
  fechaCompra!: string;  // fecha_compra en BD
  fechaVence!: string;   // fecha_vence en BD
  fechaCreacionRegistro?: string;  // fecha_creacion_registro en BD
  fechaModificacionRegistro?: string;  // fecha_modificacion_registro en BD
}

File: src/main/java/com/uniminuto/clinica/entity/Medicamento.java
package com.uniminuto.clinica.entity;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "medicamento")
public class Medicamento implements Serializable {
    /**
     * Id serializable.
     */
    private static final long serialVersionUID = 1L;

    /**
     * Id.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "descripcion")
    private String descripcion;

    @Column(name = "presentacion")
    private String presentacion;

    @Column(name = "fecha_compra")
    private LocalDate fechaCompra;

    @Column(name = "fecha_vence")
    private LocalDate fechaVence;

    @Column(name = "fecha_creacion_registro")
    private LocalDateTime fechaCreacionRegistro;

    @Column(name = "fecha_modificacion_registro")
    private LocalDateTime fechaModificacionRegistro;
}
